gobblin-hive-registration/src/main/java/org/apache/gobblin/hive/metastore/HiveMetaStoreBasedRegister.java

```diff
@@ -501,8 +482,30 @@ private void addOrAlterPartition(IMetaStoreClient client, Table table, HiveParti
       }
     }
   }
+  protected void OnPartitionExist(IMetaStoreClient client, Table table, HivePartition partition, Partition nativePartition, TException e) throws TException {
```

Comment:
Why not make `addOrAlterPartition` protected and overwrite that ?

Reply:
Because most of the logic are the same, ie. checking the key value size and use lock to call add_partition method. The only different is after it found the partition has already existed.